test/NuGet.Core.Tests/NuGet.Packaging.Test/NoRefOrLibFolderInPackageRuleTests.cs

```diff
@@ -0,0 +1,77 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using NuGet.Common;
+using NuGet.Commands;
+using NuGet.Test.Utility;
+using Xunit;
+using NuGet.Packaging.Rules;
+using Moq;
+using System.Xml.Linq;
+using System.Collections;
+using System.Runtime.CompilerServices;
+using System.Reflection;
+using System.Runtime.InteropServices.ComTypes;
+
+namespace NuGet.Packaging.Test
+{
+    public class NoRefOrLibFolderInPackageRuleTests
+    {
+        [Theory]
+        [MemberData("WarningRaisedData", MemberType = typeof(FileSource))]
+        public void WarningRaisedWhenLibOrRefFolderWithTFMDataIsNotFoundAndBuildFolderIsFound(string[] files)
+        {
+            //Act
+            var rule = new NoRefOrLibFolderInPackageRule(AnalysisResources.NoRefOrLibFolderInPackage);
+            var issues = rule.Validate(files).ToList();
+
+            // Assert
+            Assert.True(issues.Any(p => p.Code == NuGetLogCode.NU5127));
+        }
+
+        [Theory]
+        [MemberData("WarningNotRaisedData", MemberType = typeof(FileSource))]
+        public void WarningNotRaisedWhenLibOrRefFolderWithTFMDataIsFound(string[] files)
+        {
+            //Act
+            var rule = new NoRefOrLibFolderInPackageRule(AnalysisResources.NoRefOrLibFolderInPackage);
+            var issues = rule.Validate(files).ToList();
+
+            // Assert
+            Assert.False(issues.Any(p => p.Code == NuGetLogCode.NU5127));
+        }
+
+        public static class FileSource
+        {
+            public static readonly List<object[]> WarningNotRaisedData
+                    = new List<object[]>
+                    {
+                    new object[] { new string[] {"build/net45/test.props", "ref/net45/test.dll"} },
```

Comment:
What would happen if 

`build/net45/test.props, ref/test.dll` is the input?

Reply:
That should be a different rule altogether. We care if there is no ref folder or lib folder. Anything else will not raise this warning